### DISCUSSION TITLE
Feedback - Where would you draw the line when using Alpine?

### DISCUSSION THREAD

**Discussion by Vad1mo** [alpinejs/alpine#1173] [1. Help] [answered]
Hello, 
I mostly worked on backend site, but I know angular(3/10), vuejs(6/10) and alpine(8/10).

In my last project (GoHugo static website) I selected alpine to sprinkle a bit of logic here and there. On forms and dynamic toggles. Nothing that requires a REST API. 

As things progress, we now want to implement some more logic with a backend. We want to basically have form that allows users to invite other users to projects. Super simple and alpine would be a good fit. But that user needs to be authenticated as we only permit a user to invite other users to project they are part of themselves. 

We already use [auth0.js](https://github.com/auth0/auth0.js) in a different context and would like to use it as well for this tiny flow. 

So I wonder if it's still a good idea to use alpine for that or maybe switch to vue. 

Or asked differently, when would you draw a line and say this job is better done with vue/react and friends?

**Comment by HugoDF** [comment, marked as answer]
Alpine handles AJAX requests fine (using the fetch Web API or a client like axios).

I personally would draw the line at SPAs, Alpine doesn't have a routing mechanism for example where Vue/React do have packages for that.

On the topic of packages, that's another good reason to use React/Vue, Alpine doesn't have Auth0 components/React hooks and/or specific integration docs.

And that's the final place where other tools still have an edge: documentation & examples. Since Vue/React/etc are more widely adopted, a lot more has been written about all sorts of different combinations (eg. Vue + Auth0) than with Alpine & you'll be able to ask a larger community for help (although a few of us try to be active in the Discord).

Note: this is my opinion on this, not official Alpine recommendations
